be/src/storage/rowset/rowset_factory.cpp

```diff
@@ -36,7 +36,7 @@ OLAPStatus RowsetFactory::create_rowset(MemTracker* mem_tracker, const TabletSch
                                         const std::string& rowset_path, const RowsetMetaSharedPtr& rowset_meta,
                                         RowsetSharedPtr* rowset) {
     if (rowset_meta->rowset_type() == BETA_ROWSET) {
-        rowset->reset(new BetaRowset(mem_tracker, schema, rowset_path, rowset_meta));
+        *rowset = std::make_shared<BetaRowset>(mem_tracker, schema, rowset_path, rowset_meta);
```

Comment:
What is the difference between reset and assignment of std::shared_ptr in these cases?